rdmo/projects/progress.py

```diff
@@ -25,17 +27,85 @@ def get_catalog_conditions(catalog):
     ).distinct().select_related('source', 'target_option')
 
 
-def resolve_conditions(catalog, values, sets):
+def _prefetch_tree_conditions(catalog):
```

Comment:
I think instead of `_prefetch_tree_conditions` `Catalog.prefetch_elements` could be used which is already existing and more readable since it does not use recursion.